monitor/avs/monitor.go

```diff
@@ -13,58 +14,146 @@ import (
+	operatorStakeGuage.Reset()
+	operatorDelegationsGuage.Reset()
 
 	var total float64
 	for _, operator := range operators {
 		addr := operator.Addr.Hex()
 		staked := weiToEth(operator.Staked)
 		delegated := weiToEth(operator.Delegated)
 
-		operatorStake.WithLabelValues(addr).Set(staked)
-		operatorDelegations.WithLabelValues(addr).Set(delegated)
+		operatorStakeGuage.WithLabelValues(addr).Set(staked)
+		operatorDelegationsGuage.WithLabelValues(addr).Set(delegated)
 
 		total += delegated + staked
 	}
 
-	numOperators.Set(float64(len(operators)))
-	totalDelegations.Set(total)
+	numOperatorsGuage.Set(float64(len(operators)))
+	totalDelegationsGuage.Set(total)
+
+	return nil
+}
+
+func monitorOwnerOnce(avs *bindings.OmniAVS) error {
+	owner, err := avs.Owner(&bind.CallOpts{})
+	if err != nil {
+		return errors.Wrap(err, "get owner")
+	}
+
+	ownerGauge.WithLabelValues(owner.Hex()).Set(1)
+
+	return nil
+}
+
+func monitorAllowlistEnabledOnce(avs *bindings.OmniAVS) error {
+	enababled, err := avs.AllowlistEnabled(&bind.CallOpts{})
+	if err != nil {
+		return errors.Wrap(err, "get allowlist")
+	}
+
+	if enababled {
+		allowlistEnabledGuage.Set(1)
+	} else {
+		allowlistEnabledGuage.Set(0)
+	}
+
+	return nil
+}
+
+func monitorPausedOnce(avs *bindings.OmniAVS) error {
+	paused, err := avs.Paused(&bind.CallOpts{})
+	if err != nil {
+		return errors.Wrap(err, "get paused")
+	}
+
+	if paused {
+		pausedGuage.Set(1)
+	} else {
+		pausedGuage.Set(0)
+	}
 
 	return nil
 }
 
+func monitorMinStakeOnce(avs *bindings.OmniAVS) error {
+	stake, err := avs.MinOperatorStake(&bind.CallOpts{})
+	if err != nil {
+		return errors.Wrap(err, "get min stake")
+	}
+
+	minStakeGuage.Set(weiToEth(stake))
+
+	return nil
+}
+
+func monitorMaxOperatorsOnce(avs *bindings.OmniAVS) error {
+	max, err := avs.MaxOperatorCount(&bind.CallOpts{})
+	if err != nil {
+		return errors.Wrap(err, "get max operators")
+	}
+
+	maxOperatorsGuage.Set(float64(max))
+
+	return nil
+}
+
+func monitorStrategyParamsOnce(avs *bindings.OmniAVS) error {
+	params, err := avs.StrategyParams(&bind.CallOpts{})
+	if err != nil {
+		return errors.Wrap(err, "get strategy params")
+	}
+
+	strategyParamsGuage.Reset()
+	for _, p := range params {
+		start := p.Strategy.Hex()
+		multiplier := p.Multiplier.Int64()
+		strategyParamsGuage.WithLabelValues(start).Set(float64(multiplier))
+	}
+
+	return nil
+}
+
+type monitorOnceFunc func(avs *bindings.OmniAVS) error
+
+// monitorForever runs the given monitor function every 30 seconds until the context is canceled.
+func monitorForever(ctx context.Context, avs *bindings.OmniAVS, name string, f monitorOnceFunc) {
+	ticker := time.NewTicker(time.Second * 30)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			err := f(avs)
+			if ctx.Err() != nil {
+				return
+			} else if err != nil {
+				log.Error(ctx, fmt.Sprintf("Monitoring AVS %s failed (will retry)", name), err)
+				continue
+			}
+		}
+	}
+}
```

Comment:
```suggestion
type monitor func(ctx context.Context, avs *bindings.OmniAVS)

// monitorForever runs the given monitor function every 30 seconds until the context is canceled.
func monitorForever(ctx context.Context, avs *bindings.OmniAVS, name string, fns ...monitor) {
	ticker := time.NewTicker(time.Second * 30)
	defer ticker.Stop()

	for {
		select {
		case <-ctx.Done():
			return
		case <-ticker.C:
			for _, f := range fns {
				go f(ctx, avs)
			}
		}
	}
}
```

Reply:
So move error logging to each individual monitor? Also makes it harder to use different intervals for different monitors in future, but that's okay 30 seconds is fine for everything is what we do now

Reply:
what problem were you trying to solve from current approach? aside from just refactoring

Reply:
And would that `defer cancel()` on line 149 just cancel everything after the for loop, so right away?